### ISSUE TITLE
LinkedList.remove() makes transpiler fail

### ISSUE THREAD

**Issue by blagae** [cincheo/jsweet#517]
The transpiler fails with an `IndexOutOfBoundsException: Index: 0, Size: 0` when I call the no-args method `LinkedList.remove()`. Root cause seems to be that there is a hard-coded case statement for the method `List.remove(int idx)`.

Offending code:
```
protected boolean substituteMethodInvocationOnArray(MethodInvocationElement invocation, String targetMethodName,
			String targetClassName, boolean delegate) {
		switch (targetMethodName) {
//...
		case "remove":
		case "removeElement":
			printMacroName(targetMethodName);
			if (Util.isNumber(invocation.getArgument(0).getType())
//...
```

Client code which produces the exception:

```
public static void main(String[] args) {
    LinkedList<String> l = new LinkedList<>();
    l.add("Hello");
    l.add("world");
    l.remove(0); // no exception
    l.remove(); // exception
}
```

Stack trace:

```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at com.sun.tools.javac.util.List.get(List.java:476)
	at org.jsweet.transpiler.model.support.MethodInvocationElementSupport.getArgument(MethodInvocationElementSupport.java:63)
	at org.jsweet.transpiler.extension.RemoveJavaDependenciesAdapter.substituteMethodInvocationOnArray(RemoveJavaDependenciesAdapter.java:1158)
	at org.jsweet.transpiler.extension.RemoveJavaDependenciesAdapter.substituteMethodInvocation(RemoveJavaDependenciesAdapter.java:250)
	at org.jsweet.transpiler.Java2TypeScriptTranslator.visitApply(Java2TypeScriptTranslator.java:3536)
	at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1465)
	at org.jsweet.transpiler.util.AbstractTreeScanner.scan(AbstractTreeScanner.java:238)
	at com.sun.tools.javac.tree.TreeScanner.visitExec(TreeScanner.java:175)
	at com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1296)
	at org.jsweet.transpiler.util.AbstractTreeScanner.scan(AbstractTreeScanner.java:238)
	at org.jsweet.transpiler.util.AbstractTreePrinter.print(AbstractTreePrinter.java:125)
	at org.jsweet.transpiler.Java2TypeScriptTranslator.printBlockStatement(Java2TypeScriptTranslator.java:2887)
	at org.jsweet.transpiler.Java2TypeScriptTranslator.printBlockStatements(Java2TypeScriptTranslator.java:2880)
	at org.jsweet.transpiler.Java2TypeScriptTranslator.visitMethodDef(Java2TypeScriptTranslator.java:2474)
	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:778)
	at org.jsweet.transpiler.util.AbstractTreeScanner.scan(AbstractTreeScanner.java:238)
[etc]
```

**Comment by blagae** [comment]
Since the default implementation of `LinkedList` is this ...

```
public E remove() {
    return removeFirst();
}
```

... I figured I could also refactor my client code for the time being and use `LinkedList.removeFirst();`. However, this produces the following error:

```2019-02-21 15:59:27.027 ERROR output:55 - property 'removeFirst' does not exist on type 'string[]' at C:\ts-jsweet\src\main\java\quickstart\QuickStart.java(16)```

**Comment by blagae** [comment]
Proposed (naïve) fix:

```
		case "remove":
		case "removeElement":
			printMacroName(targetMethodName);
                        if (invocation.getArgumentCount() == 0) {
                            print(invocation.getTargetExpression(), delegate)
                                    .print(".splice(0, 1)[0]");
                        } else if (Util.isNumber(invocation.getArgument(0).getType()) // etc
```

**Comment by lgrignon** [comment]
Hello @blagae 
Thanks for your analysis which seems pretty good to me.

Could you please add a simple test case (for remove and removeFirst) in JSweet's unit tests (JUnit, simple to run :)) 
And I would be glad to review your PR, and then redeploy JSweet 2.2.0-SNAPSHOT.

**Comment by lgrignon** [comment]
We are planning to release 2.2.0 soon since it is now pretty stable.

Bye

**Comment by blagae** [comment]
Fixed by #518
